lua/nvim-tree/modified.lua

```diff
@@ -0,0 +1,34 @@
+local M = {}
+
+---@type table<string, boolean> record of which file is modified
+M._record = {}
+
+---refresh M.record
+function M.reload()
+  M._record = {}
+  local bufs = vim.fn.getbufinfo { bufmodified = true, buflisted = true }
+  for _, buf in pairs(bufs) do
+    local path = buf.name
+    M._record[path] = true
+    while path ~= vim.fn.getcwd() and path ~= "/" do
+      path = vim.fn.fnamemodify(path, ":h")
+      M._record[path] = true
+    end
+  end
+end
+
+---@param node table TODO: we should probably have a Node type
```

Comment:
We do: https://github.com/nvim-tree/nvim-tree.lua/blob/89c79cb33bc4867ccf9ee4810697da76ac59fdba/lua/nvim-tree/explorer/node-builders.lua#L13

That's one of the reasons I want to make a Node class...

Reply:
Oh I see, you meant Node class, not a type field.

In the future. I'd like to have subclasses for file, link etc. as well as a subclass for `TreeExplorer`